api/cluster_create.go

```diff
@@ -16,28 +16,115 @@ package api
 
 import (
 	"context"
+	"fmt"
 	"net/http"
 
+	clusterAPI "github.com/banzaicloud/pipeline/api/cluster"
 	"github.com/banzaicloud/pipeline/auth"
 	"github.com/banzaicloud/pipeline/cluster"
+	intCluster "github.com/banzaicloud/pipeline/internal/cluster"
 	ginutils "github.com/banzaicloud/pipeline/internal/platform/gin/utils"
 	"github.com/banzaicloud/pipeline/model/defaults"
 	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
 	pkgCommon "github.com/banzaicloud/pipeline/pkg/common"
 	"github.com/banzaicloud/pipeline/secret"
 	"github.com/gin-gonic/gin"
+	"github.com/goph/emperror"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+func (a *ClusterAPI) parseRequest(ctx *gin.Context, req interface{}) bool {
+	if err := ctx.ShouldBindJSON(req); err != nil {
+		err = emperror.Wrapf(err, "failed to parse request into %T", req)
+
+		a.errorHandler.Handle(err)
+		pkgCommon.ErrorResponseWithStatus(ctx, http.StatusBadRequest, err)
+
+		return false
+	}
+	return true
+}
+
+func isInputValidationError(err error) bool {
+	type inputValidationErrorer interface {
+		InputValidationError() bool
+	}
+
+	err = errors.Cause(err)
+	if e, ok := err.(inputValidationErrorer); ok {
+		return e.InputValidationError()
+	}
+
+	return false
+}
+
+func (a *ClusterAPI) handleCreationError(ctx *gin.Context, err error) {
+	a.errorHandler.Handle(err)
+
+	status := http.StatusInternalServerError
+	if isInputValidationError(err) {
+		status = http.StatusBadRequest
+	}
+	pkgCommon.ErrorResponseWithStatus(ctx, status, err)
+}
+
 //CreateClusterRequest gin handler
 func (a *ClusterAPI) CreateClusterRequest(c *gin.Context) {
 	a.logger.Info("Cluster creation started")
 
-	a.logger.Debug("Bind json into CreateClusterRequest struct")
-	// bind request body to struct
+	ctx := ginutils.Context(context.Background(), c)
+
+	orgID := auth.GetCurrentOrganization(c.Request).ID
+	userID := auth.GetCurrentUser(c.Request).ID
+
+	a.logger.Debug("Try to bind JSON to CreateClusterRequest struct")
```

Comment:
This log isn't necessary (or should be trace level instead). I would probably log the binding error instead (using info level, since this isn't an error from the application point of view).